nomad/config.go

```diff
@@ -535,6 +538,7 @@ func DefaultConfig() *Config {
 		DeploymentQueryRateLimit: deploymentwatcher.LimitStateQueriesPerSecond,
 		JobDefaultPriority:       structs.JobDefaultPriority,
 		JobMaxPriority:           structs.JobDefaultMaxPriority,
+		JobTrackedVersions:       structs.JobTrackedVersions,
```

Comment:
As I noted [here](https://github.com/hashicorp/nomad/pull/17939#discussion_r1268088124), the name `structs.JobTrackedVersions` should be changed to indicate it's a default value (along with its docstring):

```suggestion
		JobTrackedVersions:       structs.DefaultJobTrackedVersions,
```